okapi-core/src/test/java/org/folio/okapi/service/impl/DockerModuleHandleTest.java

```diff
@@ -87,28 +80,27 @@ public void testDockerVersionAtLocal(TestContext context) {
     DockerModuleHandle dh = new DockerModuleHandle(vertx, ld,
       "mod-users-5.0.0-SNAPSHOT", ports, "localhost", 9232, new JsonObject());
 
+    JsonObject versionRes = new JsonObject();
+    Async async = context.async();
     dh.getUrl("/version", res -> {
-      Assume.assumeTrue(res.succeeded());
-      if (res.failed()) {
-        logger.warn(res.cause().getMessage());
-        async.complete();
-        return;
+      if (res.succeeded()) {
+        versionRes.put("result", res.result());
       }
-      dh.deleteUrl("/version", "msg", res2 -> { // provoke 404 not found
-        context.assertTrue(res2.failed());
-        if (res2.failed()) {
-          context.assertTrue(res2.cause().getMessage().startsWith("msg HTTP error 404"),
-            res2.cause().getMessage());
-        }
-        dh.postUrlJson("/version", "msg", "{}", res3 -> { // provoke 404 not found
-          context.assertTrue(res2.failed());
-          if (res3.failed()) {
-            context.assertTrue(res3.cause().getMessage().startsWith("msg HTTP error 404"),
-              res3.cause().getMessage());
-          }
-          async.complete();
-        });
-      });
+      async.complete();
     });
+    async.await(1000);
+    Assume.assumeTrue(versionRes.containsKey("result"));
+    context.assertTrue(versionRes.getJsonObject("result").containsKey("Platform"));
+
+    // provoke 404 not found
+    dh.deleteUrl("/version", "msg", context.asyncAssertFailure(cause -> {
+      context.assertTrue(cause.getMessage().startsWith("msg HTTP error 404"),
+        cause.getMessage());
+      // provoke 404 not found
+      dh.postUrlJson("/version", "msg", "{}", context.asyncAssertFailure(cause2 -> {
+        context.assertTrue(cause2.getMessage().startsWith("msg HTTP error 404"),
+          cause2.getMessage());
+      }));
+    }));
```

Comment:
It seems `dh.deleteUrl` and `dh.postUrlJson` can be tested independently. Is there a reason to chain them?

Reply:
Well. they must be executed only if Docker is available through domain socket. But they could be separate anyway.

Reply:
It seems to be overkill to move check for docker in Setup.. So I'd rather keep this in one test.

Reply:
I did not mean to separate them to different tests. Below is what I was thinking, pure format change to look a bit  nicer.
```
// provoke 404 not found
dh.deleteUrl("/version", "msg", context.asyncAssertFailure(cause -> {
  context.assertTrue(cause.getMessage().startsWith("msg HTTP error 404"), cause.getMessage());
}));
// provoke 404 not found
dh.postUrlJson("/version", "msg", "{}", context.asyncAssertFailure(cause -> {
  context.assertTrue(cause.getMessage().startsWith("msg HTTP error 404"), cause.getMessage());
}));
```